pkg/config/middleware.go

```diff
@@ -0,0 +1,136 @@
+package config
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+
+	// "encoding/json"
+	"fmt"
+
+	// "log"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	authv1 "k8s.io/api/authentication/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/klog"
+)
+
+var kClientset *kubernetes.Clientset
+
+func KubeClient() string {
+	config, err, token := getClientConfig()
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+	kClientset, err = kubernetes.NewForConfig(config)
+
+	if err != nil {
+		klog.Fatal(err.Error())
+	}
+	return token
+
+}
+
+func getKubeConfigPath() string {
+	defaultKubePath := filepath.Join(os.Getenv("HOME"), ".kube", "config")
+	if _, err := os.Stat(defaultKubePath); os.IsNotExist(err) {
+		// set default to empty string if path does not reslove
+		defaultKubePath = ""
+	}
+
+	kubeConfig := getEnv("KUBECONFIG", defaultKubePath)
+	return kubeConfig
+}
+
+func getClientConfig() (*rest.Config, error, string) {
+	kubeConfigPath := getKubeConfigPath()
+	var clientConfig *rest.Config
+	var clientConfigError error
+
+	if kubeConfigPath != "" {
+		klog.Infof("Creating k8s client using KubeConfig at: %s", kubeConfigPath)
+		clientConfig, clientConfigError = clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	} else {
+		klog.V(2).Info("Creating k8s client using InClusterClientConfig")
+		clientConfig, clientConfigError = rest.InClusterConfig()
+	}
+
+	if clientConfigError != nil {
+		klog.Fatal("Error getting Kube Config: ", clientConfigError)
+	}
+
+	token := clientConfig.BearerToken
+	fmt.Printf("The bearer token is: %s\n", token)
```

Comment:
No need to return `token` because you return clientConfig, which contains the BearerToken.

**Note this is our Search API's token.**